### PR TITLE
Update OS info

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Fix OS info

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/11854
- Other reference link(s): https://github.com/pingcap/docs/pull/10595

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -18,8 +18,8 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
 
     |  Operating systems   |   Supported CPU architectures   |
     |   :---   |   :---   |
-    | Red Hat Enterprise Linux 8.4 or a later 8.x version  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
-    | CentOS 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
+    | <ul><li>Red Hat Enterprise Linux 8.4 or a later 8.x version</li><li>CentOS 8.4 or a later 8.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```
Comment (dveeden) [inline hardware-and-software-requirements.md:21]: There are two 'flavours' of CentOS 8:

- CentOS Linux 8, which has minor versions like 8.4. This has reached end-of-life. See also: https://www.centos.org/centos-linux-eol/
- CentOS Stream 8, which does not have minor versions. This is still supported upstream.

I don't think we should support a Linux distribution that is no longer supported upstream with security updates etc.

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -18,8 +18,8 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
...
+    | Red Hat Enterprise Linux 8 Stream | ARM 64 |
```
Comment (dveeden) [inline hardware-and-software-requirements.md:22]: There is no "Stream" version of RHEL. Only CentOS has a "Stream"

CentOS Stream 8 is a rolling upgrade without minor versions.
RHEL 8 is not a rolling release and has minor versions.

See also https://access.redhat.com/support/policy/updates/errata

image

**Review by dveeden — CHANGES_REQUESTED** [review]

**Comment by dveeden** [comment]
It isn't clear to me from the description what this is intended to fix, but if it is the concerns around the quality of "CentOS Stream 8" then this might be an alternative:

```patch
diff --git a/hardware-and-software-requirements.md b/hardware-and-software-requirements.md
index cb318044c..7137fd630 100644
--- a/hardware-and-software-requirements.md
+++ b/hardware-and-software-requirements.md
@@ -19,7 +19,6 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
     |  Operating systems   |   Supported CPU architectures   |
     |   :---   |   :---   |
     | Red Hat Enterprise Linux 8.4 or a later 8.x version  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
-    | CentOS 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
     | <ul><li>Red Hat Enterprise Linux 7.3 or a later 7.x version</li><li>CentOS 7.3 or a later 7.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
     | Amazon Linux 2 | <ul><li>x86_64</li><li>ARM 64</li></ul> |
     | Kylin Euler V10 SP1/SP2   |   <ul><li>x86_64</li><li>ARM 64</li></ul>   |
@@ -33,6 +32,7 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
 
     |  Operating systems   |   Supported CPU architectures   |
     |   :---   |   :---   |
+    | CentOS 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
     |   macOS Catalina or later   |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
     |  Oracle Enterprise Linux 7.3 or a later 7.x version  |  x86_64           |
     |   Ubuntu LTS 18.04 or later   |  x86_64           |
```

**Review by dveeden — APPROVED** [review]

**Comment by JasonWu0506** [comment]
LGTM

**Pull request merged**